src/pkg/cli/tail.go

```diff
@@ -31,6 +31,16 @@ var (
 	DoVerbose     = false
 )
 
+type TailDetectStopEventFunc func(service string, host string, eventlog string) bool
+
+type TailOptions struct {
+	Service            string
+	Etag               string
+	Since              time.Time
+	Raw                bool
+	EndEventDetectFunc TailDetectStopEventFunc
+}
```

Comment:
Created these as the Tail function signature was getting a bit long.
The EndEventDetectFunc is defined by the called and returns true if a specific string is detected.